### PR TITLE
Blocking interface calls

### PR DESCRIPTION
**Pull request by jadr2ddude**
This PR fixes up function typing so that the interface `$invoke` and dispatch functions can be processed by the goroutine lowering pass. As this is currently structured, these functions will never need to be turned into coroutines because all blocking calls are tail calls.

### CODE REVIEW

**Comment by jadr2ddude** [comment]
fixes #514

File: compiler/interface.go [hunk 1]
```diff
@@ -461,6 +461,7 @@ func (c *Compiler) getInterfaceInvokeWrapper(f *ir.Function) llvm.Value {
 	paramTypes := append([]llvm.Type{c.i8ptrType}, fnType.ParamTypes()[len(expandedReceiverType):]...)
 	wrapFnType := llvm.FunctionType(fnType.ReturnType(), paramTypes, false)
 	wrapper = llvm.AddFunction(c.mod, wrapperName, wrapFnType)
+	wrapper.LastParam().SetName("parentHandle")
```
Comment (aykevl) [inline compiler/interface.go]: Can you add a check whether the to-be-wrapped function also has this name? That would protect against future changes that might break this assumption.

```suggestion
	if f.LLVMFn.LastParam().Name() == "parentHandle" {
		wrapper.LastParam().SetName("parentHandle")
	}
```

**Comment by jadr2ddude** [comment]
Alright, pushed your change. Interesting that GitHub now has a button for this.

**Comment by aykevl** [comment]
Thank you! Merged in 98eee7c22ab8f790a1a712619b105a1f9a45e271.

**Pull request closed without merging**